pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -215,6 +237,22 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
               dictionaryCreator.getNumBytesPerEntry());
           throw e;
         }
+
+        if (bloomFilterColumns.contains(columnName)) {
```

Comment:
(MAJOR) bloom filter can be applied to both dictionary encoded and raw index columns.

Reply:
I'm not too sure, but looks like bloom filter only gets created for columns where dictionary encoding is enabled https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java#L142?
Also this docs says bloom filters are only supported on dictionary encoded columns?
https://docs.pinot.apache.org/basics/indexing/bloom-filter

Reply:
From the segment creator side, we can support bloom filter with/without dictionary because we use values to create the bloom filter instead of the dict ids. We don't support bloom filter on raw index currently because the handler side support is not implemented. Let's create an issue to add support on the handler side. Here we can move this outside of the if check, which should just work

Reply:
Agreed that it'll work if we just move it outside the if block, but I had intentionally put it there to ensure the behaviour remains consistent across offline segment creation, and when bloomFilter is being build during segment loading. Anyway, I've moved it out, and raised a new issue https://github.com/apache/pinot/issues/8735.